kallsyms/kallsyms.go

```diff
@@ -614,7 +704,16 @@ func (s *Symbolizer) StartMonitor(ctx context.Context) error {
 		return fmt.Errorf("failed to create kobject netlink socket: %v", err)
 	}
 	go s.reloadWorker(ctx, kobjectClient)
-	go s.pollKobjectClient(kobjectClient)
+	go s.pollKobjectClient(ctx, kobjectClient)
+	major, minor, _, err := tracer.GetCurrentKernelVersion()
```

Comment:
Since we're now calling this function multiple times, maybe we can switch to calling it once, on agent/receiver startup, perform the error check once and cache the result on success. Then the rest of the callers will always get valid version and not have to error check.

Doesn't need to happen in this PR.

Reply:
Resolving with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/1019/commits/6f86b7ae23abfefb048dfcc4e44e268f1e9eee4a as I had to resolve a cyclic dependency issue for being able to use `GetCurrentKernelVersion()`.